### PR TITLE
Move azure-arm-rest to Common directory

### PR DESCRIPTION
**Pull request by bishal-pdMSFT**

### CODE REVIEW

File: Tasks/AzureResourceGroupDeployment/task.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
     "version": {
         "Major": 2,
         "Minor": 1,
-        "Patch": 6
+        "Patch": 7
     },
     "demands": [],
     "minimumAgentVersion": "2.0.0",
```
Comment (ds-ms) [inline Tasks/AzureResourceGroupDeployment/task.json]: Remove the messages which are meant for the library

File: Tasks/Common/azure-arm-rest/Strings/resources.resjson/en-US/resources.resjson [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  "loc.messages.DeploymentNameCannotBeNull": "deploymentName cannot be null or undefined and it must be of type string.",
+  "loc.messages.CredentialsCannotBeNull": "'credentials' cannot be null.",
+  "loc.messages.SubscriptionIdCannotBeNull": "'subscriptionId' cannot be null.",
+  "loc.messages.InvalidResponseLongRunningOperation": "Invalid response received for fetching status of a long running operation.",
+  "loc.messages.TimeoutWhileWaiting": "Timed out while waiting",
+  "loc.messages.ResourceGroupCannotBeNull": "resourceGroupName cannot be null or undefined and it must be of type string.",
+  "loc.messages.ResourceGroupExceededLength": "\"resourceGroupName\" should satisfy the constraint - \"MaxLength\": 90",
+  "loc.messages.ResourceGroupDeceededLength": "\"resourceGroupName\" should satisfy the constraint - \"MinLength\": 1",
+  "loc.messages.ResourceGroupDoesntMatchPattern": "\"resourceGroupName\" should satisfy the constraint - \"Pattern\": /^[-\\w\\._\\(\\)]+$/"
+}
```
Comment (ds-ms) [inline Tasks/Common/azure-arm-rest/Strings/resources.resjson/en-US/resources.resjson]: do verify if these strings are getting picked or not, if there are duplicate keys.

File: Tasks/AzureResourceGroupDeployment/task.json (hunk 1 above)
Comment (bishal-pdMSFT) [inline Tasks/AzureResourceGroupDeployment/task.json]: I will

File: Tasks/Common/azure-arm-rest/Strings/resources.resjson/en-US/resources.resjson (hunk 2 above)
Comment (bishal-pdMSFT) [inline Tasks/Common/azure-arm-rest/Strings/resources.resjson/en-US/resources.resjson]: As per other common module, this should work. I will test after removing strings from task.

Comment (bishal-pdMSFT) [inline Tasks/Common/azure-arm-rest/Strings/resources.resjson/en-US/resources.resjson]: This does not work with module.json. I have sent a mail to Eric/Bryan asking for any work-around. In the meanwhile I will not remove these strings from task.json.

File: Tasks/AzureResourceGroupDeployment/task.json (hunk 1 above)
Comment (bishal-pdMSFT) [inline Tasks/AzureResourceGroupDeployment/task.json]: As mentioned below, I will wait for resolution on module.json before rremoving strings from here. Will take it separately.

**Pull request merged**